### PR TITLE
Redirect user to the return url instead of showing "You have already taken this test." page.

### PR DESCRIPTION
**Pull request by AndreyNegrov**
**Related to task:**  https://oat-sa.atlassian.net/browse/NSUP-32
**Description:** TAO randomly not redirecting to postbackURL after test completion
**How to test:**  
1. pass test through LTI. 
2. run the test again with custom parameter '**force_redirect_to_return_url**'
image

After that you will be redirected to the 'Return URL' page

### CODE REVIEW

**Review by tikhanovichA — APPROVED** [review]
**Review Checklist**
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] The feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

File: controller/DeliveryTool.php [hunk 1]
```diff
@@ -61,6 +62,14 @@ class DeliveryTool extends ToolModule
      * @var string
      */
     const PARAM_SKIP_THANKYOU = 'custom_skip_thankyou';
+
+    /**
+     * Setting this parameter to 'true' will prevent the 'You have already taken this test'
+     * screen to be shown skip directly to the return url
+     * @var string
+     */
+    const FORCE_REDIRECT_TO_RETURN_URL = 'custom_force_redirect_to_return_url';
```
Comment (augustas) [inline controller/DeliveryTool.php:71]: It seems that the convention for constants are `PARAM_*`. And the name of parameter seems misleading for me, it does not force much, f.e. if test taker never started the delivery then there is no redirect. I'd suggest `PARAM_SKIP_OVERVIEW`

File: controller/DeliveryTool.php [hunk 2]
```diff
@@ -193,14 +203,33 @@ protected function getLearnerUrl(\core_kernel_classes_Resource $delivery, Delive
...
                 __('User is not authorized to run this delivery'),
                 LtiErrorMessage::ERROR_LAUNCH_FORBIDDEN
             );
         }
+
+        if ($user->getLaunchData()->hasVariable(self::FORCE_REDIRECT_TO_RETURN_URL)) {
+            $executionService = $this->getServiceLocator()->get(LtiDeliveryExecutionService::SERVICE_ID);
+            $executions = $executionService->getLinkedDeliveryExecutions($delivery, $currentSession->getLtiLinkResource(), $user->getIdentifier());
+            $lastDE = end($executions);
+            $url = _url(
```
Comment (augustas) [inline controller/DeliveryTool.php:218]: I think it would be better to use the `LtiNavigationService::getReturnUrl(...)` to build the return url

**Review by augustas — CHANGES_REQUESTED** [review]
Works on my machine, a few changes suggested.

**Review by augustas — APPROVED** [review]
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**